feat: Move onboarding into the app

xrdt opened the pull request:
## Problem

Ref [Cory's mocks](https://github.com/PostHog/posthog/issues/19509)

## Changes


https://github.com/PostHog/posthog/assets/21014901/1f5d7752-ee99-4e92-be54-fc248baf740f



👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

Commit:
bring onboarding panels into the app

Commit:
tune up product intro page breadcrumbs

Commit:
combine verification and config steps

Commit:
add breadcrumbs, tune up topbar breadcrumbs

Commit:
tune up some buttons, step order

Commit:
mostly update the plans page

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
tune up state for not first product onboarding

xrdt commented:
Remaining things: 
- How to do update the state so people are by default navigated to the step after Install if it the product they select isn't the first they're onboarding to.
  - There's an issue here with when the onboardingSteps get updated from [] on the intro page to whatever they actually are when navigate to the onboarding scene. 
- Allow people to toggle the plans comparison and selector open even after they've subscribed. 
- Show the upgrade without add ons button.

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
some type fixes

Commit:
fix up remaining tasks

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
make product intro onboarding step

* Only report onboarding product selected when "get started button clicked"
* Don't show intro if coming from /products page

Commit:
show verification on install screen

xrdt commented:
@raquelmsmith addressed your comments about making the intro page an actual onboarding step, updated when we fire the `onboarding product selected event` so it doesn't fire on intro page visits, and moved verification into the sdk / install page.

xrdt commented on frontend/__snapshots__/posthog-3000-navigation--navigation-base--dark.png:
These snapshots shouldn't be changing ... I updated the left side nav bar to be the correct order. Maybe it's the type error.

Commit:
fix a type error, update intro in stories

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
fix a type generation issue

raquelmsmith committed:
improve verify section

raquelmsmith committed:
sentence case

raquelmsmith committed:
make breadcrumbs smaller

raquelmsmith committed:
fix intro pages

raquelmsmith committed:
fix breadcrumbs

raquelmsmith commented:
I pushed some commits here:

1. Improved the verify bit to make it its own section:
![image](https://github.com/PostHog/posthog/assets/18598166/49631dbb-ffc4-487d-8dd8-b5d56a21585a)
![image](https://github.com/PostHog/posthog/assets/18598166/be7924de-1db2-4ac9-84b3-c25e03590799)
![image](https://github.com/PostHog/posthog/assets/18598166/0426383a-f8e3-43c2-a92f-d0333891433b)

2. The intro pages has weird margins all around, and the continue button at the bottom. 
3. The breadcrumbs didn't match the [mocks](https://github.com/PostHog/posthog/issues/19509#issue-2054545813)
4. Seems like you introduced a `.scss` file but we should really avoid that in favor of utility classes unless absolutely necessary.

There are still a few things that need fixing:
1. I removed the "Other products" step in another PR, but it still exists here, so conflicts need to be resolved.
2. The scene redirection that takes you to the product intro page no longer works.
3. I'm pretty concerned about changing the upgrade UI, I think we should do that in a different test, and cory is working on a different UI right now anyway that incorporates plans better soooo I think we shouldn't change this much right now (except maybe hiding full tier info if we want)

Commit:
update padding on onboarding steps, remove other products step

Commit:
mostly revert billing page changes?

Commit:
highlight the navbar item

Commit:
Merge remote-tracking branch 'origin/master' into by/onboarding-in-app

xrdt commented:
@raquelmsmith Latest set of changes: 
* I ended up reverting the width constraint on the non product intro onboarding pages. It feels a bit wide. If we do want to keep it to max-w-screen-lg, we can add auto-margins left and right to center the content.  
* The billing page is _mostly_ reverted? Not sure if we do want to adjust the background color / border for contrast
  * I added a header for the product feature and changed the "included platform features" header styling. 
* The left nav button should be correctly highlighted now. 
* I wasn't seeing any navigation issues (can't get back to product intro page) but let me know if there's a repro. 

* I still don't know what's going on with the screenshots. Will take a look.
https://github.com/PostHog/posthog/assets/21014901/1ad3a628-f990-40cc-a600-22933ec3b54b

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Commit:
fix type errors

Commit:
Revert "Update UI snapshots for `chromium` (1)"

This reverts commit 7b9ac964d8202fbdf7af74296116bd897fb421a5.

Commit:
Revert "Update UI snapshots for `chromium` (2)"

This reverts commit 3c4b26f6765fc5c7640f7f02c421d0495ff3ddf1.

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

marandaneto commented:
That looks much nicer and better, well done.

raquelmsmith commented:
> I ended up reverting the width constraint on the non product intro onboarding pages. It feels a bit wide. If we do want to keep it to max-w-screen-lg, we can add auto-margins left and right to center the content.

Yeah it definitely needs to be constrained. On a wider screen it just keeps getting wider forever. Generally content you read (eg a paragraph) should never exceed 700px.

Commit:
constrain width of onboarding steps, remove new tag from notebooks nav

xrdt commented:
Makes sense. I've constrained the width and centered it. 

<img width="1286" alt="Screenshot 2024-02-14 at 9 18 31 AM" src="https://github.com/PostHog/posthog/assets/21014901/d76da0b5-f98a-4c4e-be4f-eaf52f289d13">

Commit:
Merge remote-tracking branch 'origin/master' into by/onboarding-in-app

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

Commit:
fix some onboarding snapshots

Commit:
Revert "Update UI snapshots for `chromium` (2)"

This reverts commit c2fb465e306c40f2b4a82fa4c9fe7cc40dbb245b.

Commit:
Revert "Update UI snapshots for `chromium` (1)"

This reverts commit 613b5a1746cee9fe360c02aafc7280cc4baf8d64.

raquelmsmith commented on frontend/src/scenes/billing/PlanComparison.tsx:
```diff
@@ -109,44 +108,9 @@ export const PlanComparison = ({
+    const { billing } = useValues(billingLogic)
     const { width, ref: planComparisonRef } = useResizeObserver()
 
-    const upgradeButtons = plans?.map((plan) => {
```
These buttons were also used on the billing page in the plan comparisons, seems like we should keep them there?

xrdt replied:
Great point, I'll add them back.

raquelmsmith commented on frontend/src/scenes/onboarding/onboardingLogic.tsx:
```diff
@@ -18,14 +19,44 @@ export interface OnboardingLogicProps {
-    SDKS = 'sdks',
-    BILLING = 'billing',
-    OTHER_PRODUCTS = 'other_products',
+    INSTALL = 'install',
+    PLANS = 'plans',
```
Changing these means our insights will change because they filter on URL, might want to make an action to join them together. Or just have a different enum that maps the step key to a proper name.

xrdt replied:
Action for install step: https://us.posthog.com/project/2/data-management/actions/66284

Action for billing step: https://us.posthog.com/project/2/data-management/actions/66285

xrdt replied:
PA Billing: https://us.posthog.com/project/2/data-management/actions/66286
SR Billing: https://us.posthog.com/project/2/data-management/actions/66287
Surveys Biling: https://us.posthog.com/project/2/data-management/actions/66288
FF Billing: https://us.posthog.com/project/2/data-management/actions/66289

xrdt replied:
Created product specific ones for install step as well.

raquelmsmith commented on frontend/src/scenes/onboarding/OnboardingBillingStep.tsx:
```diff
@@ -40,20 +43,21 @@ export const OnboardingBillingStep = ({
                         }}
                     >
-                        Subscribe
+                        Subscribe to Paid Plan
```
No need to do this now, but this feels like a good A/B test to run- subscribe vs upgrade vs add credit card

xrdt replied:
Added to kanban

raquelmsmith commented on frontend/src/scenes/onboarding/OnboardingBillingStep.tsx:
```diff
@@ -67,6 +71,9 @@ export const OnboardingBillingStep = ({
                                 </div>
                             </div>
+                            <LemonButton className="mt-2" onClick={() => setShowPlanComp(!showPlanComp)}>
+                                {showPlanComp ? 'Hide' : 'See'} Plans
```
```suggestion
                                {showPlanComp ? 'Hide' : 'Show'} plans
```

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

raquelmsmith commented on frontend/src/scenes/onboarding/onboardingLogic.tsx:
```diff
@@ -117,6 +172,12 @@ export const onboardingLogic = kea<onboardingLogicType>([
             (s) => [s.allOnboardingSteps],
             (allOnboardingSteps: AllOnboardingSteps) => allOnboardingSteps.length,
         ],
+        onboardingStepNames: [
```
This technically doesn't get the step _name_, it gets the step _key_

xrdt commented on frontend/src/scenes/billing/PlanComparison.scss:
```diff
@@ -5,6 +5,9 @@
 
+    // background-color: var(--glass-bg-3000);
+    // border: 1px solid var(--border-bold-3000);
+
```
Thanks, I think I pushed the change to remove it just as you submitted your review.

xrdt committed:
Update frontend/src/scenes/onboarding/OnboardingBillingStep.tsx

Co-authored-by: Raquel Smith <raquelmsmith@users.noreply.github.com>

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

Commit:
wip

Commit:
address pr comments

* add back subscribe buttons on billing page
* rename a selector

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

Commit:
small change to upgrade button

Commit:
remove extraneous files

Commit:
Merge remote-tracking branch 'origin/master' into by/onboarding-in-app

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)
